lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1155,9 +1155,9 @@ private function getCommitOrder(array $entityChangeSet = null)
                 }
 
                 $joinColumns = reset($assoc['joinColumns']);
-                $isNullable  = isset($joinColumns['nullable']) ? $joinColumns['nullable'] : false;
+                $weight  = (int)empty($joinColumns['nullable']);
```

Comment:
Consider inlining this into `$calc->addDependency(`, since the variable isn't used anywhere else.